lib/ansible/modules/hostname.py

```diff
@@ -539,41 +411,27 @@ def get_permanent_hostname(self):
             except IOError as e:
```

Comment:
This still does a write, right above this line. This means we're modifying the system in this function (and this function gets called in check_mode, see #66432).

But fixing this leads down a rabbit hole of other issues:
* We don't currently standardize on "UNKNOWN" vs "" when the file doesn't exist.
  * Some modules write it (again - breaking check_mode's promise) as "" and then return that as the "before" value. 
  * Others default to "UNKNOWN" if it doesn't exist.
  * FreeBSDStrategy defaults to "temporarystub" if it doesn't exist.

My proposal (#70828) is to unify them all to `""` - but this is a breaking change if anyone depends on the "before" value.